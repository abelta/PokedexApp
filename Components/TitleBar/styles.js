import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  titleBar: {
    backgroundColor: 'blue',
    flexDirection: 'row',
  },
  title: {
    flex: 1,
    paddingLeft: 4,
    paddingRight: 4,
  },
  search: {
    flex: 1,
    justifyContent: 'center',
    paddingLeft: 4,
    paddingRight: 4,
  },
});

export default styles;
