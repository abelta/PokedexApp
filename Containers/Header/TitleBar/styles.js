import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  titleBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  empty: {
    width: 25,
  },
  title: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    paddingLeft: 4,
    paddingRight: 4,
  },
  search: {
    justifyContent: 'center',
    paddingLeft: 4,
    paddingRight: 4,
    width: 25,
  },
});

export default styles;
