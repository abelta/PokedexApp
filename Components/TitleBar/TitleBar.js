import React from 'react';
import { Text, View } from 'react-native';
import { Link } from 'react-router-native';
import { AppTitle } from '../../Components';
import styles from './styles';

const TitleBar = ({ style }) => (
  <View style={[styles.header, style]}>
    <Link to="/"><AppTitle style={style.appTitle}>Pok√©dex</AppTitle></Link>
    <Link to="/search"><Text>Search</Text></Link>
  </View>
);

TitleBar.propTypes = {
  style: View.propTypes.style,
};

TitleBar.defaultProps = {
  style: {},
};

export default TitleBar;
