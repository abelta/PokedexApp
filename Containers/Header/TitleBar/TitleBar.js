import React from 'react';
import { View } from 'react-native';
import { AppTitle, TouchableLink } from '../../../Components';
import { Search as SearchIcon } from '../../../Components/Icons';
import styles from './styles';

const TitleBar = () => (
  <View style={styles.titleBar}>
    <TouchableLink style={styles.title} to="/">
      <AppTitle>Pok√©dex</AppTitle>
    </TouchableLink>
    <TouchableLink style={styles.search} to="/search">
      <SearchIcon />
    </TouchableLink>
  </View>
);

export default TitleBar;
