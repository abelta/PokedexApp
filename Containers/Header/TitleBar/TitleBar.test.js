import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-native';
import TitleBar from '.';

jest.mock('../../../Components/SlideView');

describe('TitleBar', () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <TitleBar />
    </MemoryRouter>,
  ).toJSON();

  it('renders properly', () => {
    expect(tree).toMatchSnapshot();
  });
});
