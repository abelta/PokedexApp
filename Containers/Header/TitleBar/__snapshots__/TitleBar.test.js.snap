// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar renders properly 1`] = `null`;
